src/main/java/com/fakng/fakngagrgtr/parser/uber/UberParser.java

```diff
@@ -0,0 +1,84 @@
+package com.fakng.fakngagrgtr.parser.uber;
+
+import com.fakng.fakngagrgtr.parser.ApiParser;
+import com.fakng.fakngagrgtr.parser.LocationProcessor;
+import com.fakng.fakngagrgtr.persistent.company.CompanyRepository;
+import com.fakng.fakngagrgtr.persistent.vacancy.Vacancy;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import javax.annotation.PostConstruct;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Map;
+
+@Component
+public class UberParser extends ApiParser {
+
+    private static final String VACANCY_URL_FORMAT = "https://www.uber.com/global/en/careers/list/%s/";
+    private static final String CSRF_TOKEN_HEADER = "X-Csrf-Token";
+
+    public UberParser(WebClient webClient, CompanyRepository companyRepository, LocationProcessor locationProcessor,
+                      @Value("${url.uber}") String url) {
+        super(webClient, companyRepository, locationProcessor);
+        this.url = url;
+    }
+
+    @PostConstruct
+    public void init() {
+        initBase();
+    }
+
+    @Override
+    protected List<Vacancy> getAllVacancies() {
+        return requestVacancies().stream()
+                .map(this::createVacancy)
+                .toList();
+    }
+
+    private Vacancy createVacancy(VacancyDTO dto) {
+        Vacancy vacancy = new Vacancy();
+        vacancy.setJobId(dto.getId());
+        vacancy.setTitle(dto.getTitle());
+        vacancy.setDescription(dto.getDescription());
+        vacancy.setUrl(String.format(VACANCY_URL_FORMAT, vacancy.getJobId()));
+        vacancy.setCompany(company);
+        vacancy.setPublishedDate(parseDate(dto.getUpdatedDate()));
+        parseLocations(vacancy, dto);
+        return vacancy;
+    }
+
+    private void parseLocations(Vacancy vacancy, VacancyDTO dto) {
```

Comment:
As far as I know, we aim to make methods with 1 parameter. I see this method as:
public List<Location> parseLocations(List<LocationDTO>){...}

Accordingly, in createVacancy it will be like:
vacancy.setLocation(parseLocations(dto.getLocations()));

What do you think about it?

Reply:
Agree with Alexsei. Method parseLocation shouldn't make anything else, it literally should only parse locations and returns them. Or we should just rename it and allow change our vacancy.